changelog.d/13958.bugfix

```diff
@@ -0,0 +1 @@
+Fix performance bottleneck with heavy appservice and bridged users in Synapse by checking appservice user interest against the local users instead of all users.
```

Comment:
For anyone curious, there might be a few more `get_users_in_room` mis-uses in the `sync` and `presence` code for someone who has more domain knowledge on what those things should exactly do.